ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -369,6 +374,15 @@ class SimpleOperationTracker implements OperationTracker {
         originatingDcName);
   }
 
+  /**
+   * @param eligibleReplicaCount eligible replicas to send Delete Requests.
+   * @return true if it's possible to run offline repair for the Delete
+   */
+  public boolean possibleRunOfflineRepair(int eligibleReplicaCount) {
+    return (eligibleReplicaCount >= 1 && routerOperation == RouterOperation.DeleteOperation
+        && routerConfig.routerDeleteOfflineRepairEnabled && routerConfig.routerRepairRequestsDbFactory != null);
```

Comment:
as long as the offline repair is on, we should allow it, we don't have to check the DbFactory.

Reply:
Remove the routerRepairRequestsDBFactory check.
At the same time, make routerDeleteOfflineRepairEnabled check the factory already.